### PR TITLE
Revives Organ Datumization

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Revives #22578 . Most of this description will be fully copypasted from the original PR, as I agree with it. This is a PR without the balance change of slimes having a core inside their chest. Slimes still only have only a core, and their heart/lungs have been thrown out.

Old PR text:
Lays the groundwork for organ datumization with /datum/organ. The most recently inserted organ will take priority for that datum organ function. If an organ is removed say with lung functionality, it will check other organs for lung functionality and replace itself.

Slime people no longer have a heart and lungs, the function of those have been incorporated into the core itself. This means it handles the functions of the heart, lung, and brain all in one. I'm not including eyes and ears in this datumization, because slimes somehow function without them.

If balance team doesnt like the removal of the organs from the slimes, I can remove that part from the PR

## Why It's Good For The Game
#11990 was originally made because you needed lungs and hearts for newcrit, this PR changes that and returns them to their old single-core form. I like the old form, because it was more unique and gave slime people more flavor.

## Images of changes
(Yes I checked this for myself, I am just lazy and stole this screenshot)
image

## Testing
I've replicated every testing step from the original PR.

## Changelog
:cl: Contrabang, DGamerL
tweak: Slimepeople now don't have lungs or a heart anymore, the core inside their head now does those functions
del: Slimepeople lungs and heart
/:cl:

### CODE REVIEW

File: code/datums/diseases/_MobProcs.dm [hunk 1]
```diff
@@ -154,8 +154,13 @@
 		return FALSE
 
 	for(var/thing in D.required_organs)
```
Comment (Sirryan2002) [inline code/datums/diseases/_MobProcs.dm:156]: please for the love of god rename this variable

File: code/modules/mob/living/carbon/human/human_life.dm [hunk 2]
```diff
@@ -131,31 +131,30 @@
...
+		var/datum/species/S = dna.species
+
+		if(S.breathid == "o2")
+			throw_alert("not_enough_oxy", /obj/screen/alert/not_enough_oxy)
+		else if(S.breathid == "tox")
+			throw_alert("not_enough_tox", /obj/screen/alert/not_enough_tox)
+		else if(S.breathid == "co2") // currently unused
+			throw_alert("not_enough_co2", /obj/screen/alert/not_enough_co2)
+		else if(S.breathid == "n2")
+			throw_alert("not_enough_nitro", /obj/screen/alert/not_enough_nitro)
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/human/human_life.dm:156]: good candidate for a `switch(S.breathid)` statement

File: code/modules/surgery/organs/brain.dm [hunk 3]
```diff
@@ -187,13 +187,14 @@
...
 
+// Hello I am from the ministry of rubber forehead aliens how are you
 /obj/item/organ/internal/brain/slime
 	name = "slime core"
 	desc = "A complex, organic knot of jelly and crystalline particles."
 	icon = 'icons/mob/slimes.dmi'
 	icon_state = "green slime extract"
 	mmi_icon_state = "slime_mmi"
-//	parent_organ = "chest" Hello I am from the ministry of rubber forehead aliens how are you
+	organ_datums = list(/datum/organ/heart, /datum/organ/lungs)
```
Comment (Sirryan2002) [inline code/modules/surgery/organs/brain.dm:197]: you are not using the defines from earlier?

File: code/modules/surgery/organs/lungs.dm [hunk 4]
```diff
@@ -356,38 +47,30 @@
...
-			safe_oxygen_min = initial(safe_oxygen_min)
-			safe_toxins_min = initial(safe_toxins_min)
-			safe_toxins_max = initial(safe_toxins_max)
-			to_chat(user, "<span class='notice'>You configure [src] back to default settings.</span>")
-			species_state = "human"
+
+	var/possible = list("default" = /datum/organ/lungs, "vox" = /datum/organ/lungs/vox, "plasmamen" = /datum/organ/lungs/plasmamen)
+	var/chosen = input(user, "Select lung type", "What kind of lung settings?") as null|anything in possible
+	if(isnull(chosen) || chosen == species_state)
+		return
```
Comment (Sirryan2002) [inline code/modules/surgery/organs/lungs.dm:54]: you need to proximity check after chosen and also make sure they still have the multitool !!!

File: code/modules/mob/living/carbon/human/human_life.dm (hunk 2 above)
Comment (DGamerL) [reply code/modules/mob/living/carbon/human/human_life.dm:156]: Done

File: code/datums/diseases/_MobProcs.dm (hunk 1 above)
Comment (DGamerL) [reply code/datums/diseases/_MobProcs.dm:156]: Done!

File: code/modules/surgery/organs/brain.dm (hunk 3 above)
Comment (DGamerL) [reply code/modules/surgery/organs/brain.dm:197]: Not sure if that'll work as intended, seeing the defines are
```
#define ORGAN_DATUM_HEART	"heart"
#define ORGAN_DATUM_LUNGS	"lungs"
```
While these are the actual datums that need to be added

Comment (Sirryan2002) [reply code/modules/surgery/organs/brain.dm:197]: oh

File: code/modules/surgery/organs/lungs.dm (hunk 4 above)
Comment (DGamerL) [reply code/modules/surgery/organs/lungs.dm:54]: Yeah just need to do this, other reviews are done but unpushed

Comment (DGamerL) [reply code/modules/surgery/organs/lungs.dm]: Proximity check done, how would you suggest the multitool check? Just another `I.use_tool(src, user, 0, volume = I.tool_volume)`?

Comment (Sirryan2002) [reply code/modules/surgery/organs/lungs.dm]: yeah sure

**Review by Sirryan2002 — APPROVED** [review]
just the use_tool check and this is goo

File: code/modules/surgery/organs/lungs.dm (hunk 4 above)
Comment (DGamerL) [reply code/modules/surgery/organs/lungs.dm]: Done

**Review by S34NW — APPROVED** [review]

**Pull request merged**